hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMAllocateBlockRequest.java

```diff
@@ -227,6 +232,18 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     } catch (IOException ex) {
       omMetrics.incNumBlockAllocateCallFails();
       exception = ex;
+      if (exception.toString().contains(
+          OMException.ResultCodes.QUOTA_EXCEEDED.toString())) {
+        long keyAllocatedSpace = openKeyInfo.getLatestVersionLocations()
+            .getLocationListCount() * ozoneManager.getScmBlockSize()
+            * openKeyInfo.getFactor().getNumber();
+        // Update usedBytes atomically. ErrorOMResponse does not persist the DB,
+        // so we update the cache first. The next time another key is written,
+        // volume Args will be persisted to the DB.
+        // TODO: There is a delay in updating DB in this way, and if necessary
+        //  we can modify the ErrorOMResponse to avoid it.
+        omVolumeArgs.getUsedBytes().add(-keyAllocatedSpace);
```

Comment:
If A object has multiple blocks, and quota_exceeded exception is thrown out at the last block, shall we only deduct the block space size at this step?